test/test_prime_sieve.cpp

```diff
@@ -0,0 +1,218 @@
+    ref = 78498; // Calculated with wolfram-alpha
+    primes.clear();
+    boost::math::prime_range(static_cast<Integer>(2), static_cast<Integer>(1000000), primes);
+    BOOST_TEST_EQ(primes.size(), ref);
+}
+
+template<typename Integer>
+void test_prime_sieve_overflow()
+{
+    std::vector<Integer> primes;
+
+    // Should die with call to BOOST_ASSERT
+    boost::math::prime_sieve(static_cast<Integer>(2), static_cast<Integer>(std::numeric_limits<Integer>::max()), primes);
+}
+
+template<typename Integer>
+void test_par_prime_sieve_large()
+{
+    std::vector<Integer> primes;
+    Integer ref {54400028}; // Calculated with wolfram-alpha
+
+    // Force the sieve into the multi-threading section and test reserve functionality
+    boost::math::prime_reserve(static_cast<Integer>(1073741824), primes);
+    boost::math::prime_sieve(static_cast<Integer>(1073741824), primes);
+    BOOST_TEST_EQ(primes.size(), ref);
+}
+
+template<typename Integer>
+void test_interval_sieve()
+{
+    std::vector<Integer> pre_sieved_primes {2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71};
+    std::vector<Integer> primes;
+
+    boost::math::detail::IntervalSieve sieve(1'000, 10'000, pre_sieved_primes, primes);
+    BOOST_TEST_EQ(primes.size(), 1'061);
+
+    primes.clear();
+    sieve.NewRange(10'000, 100'000, primes);
+    BOOST_TEST_EQ(primes.size(), 8'363);
+
+    primes.clear();
+    sieve.NewRange(100'000, 1'000'000, primes);
+    BOOST_TEST_EQ(primes.size(), 68'906);
+}
+
+template<typename Integer>
+void test_linear_sieve()
+{
+    std::vector<Integer> primes;
+
+    boost::math::detail::linear_sieve(static_cast<Integer>(1'000), primes);
+    BOOST_TEST_EQ(primes.size(), 168);
+
+    primes.clear();
+    boost::math::detail::linear_sieve(static_cast<Integer>(10'000), primes);
+    BOOST_TEST_EQ(primes.size(), 1229);
+
+    primes.clear();
+    boost::math::detail::linear_sieve(static_cast<Integer>(100'000), primes);
+    BOOST_TEST_EQ(primes.size(), 9592);
+}
+
+int main()
+{
+    // Individual Algorithms
+    //test_linear_sieve<int>();
+    //test_linear_sieve<int32_t>();
+    test_linear_sieve<int64_t>();
+    //test_linear_sieve<uint32_t>();
+    
+    //test_interval_sieve<int>();
+    //test_interval_sieve<int32_t>();
+    test_interval_sieve<int64_t>();
+    //test_interval_sieve<uint32_t>();
+
+    // Composite
+    //test_prime_sieve<int>();
+    //test_prime_sieve<int32_t>();
+    test_prime_sieve<int64_t>();
+    //test_prime_sieve<uint32_t>();
+    //test_prime_sieve<boost::multiprecision::cpp_int>();
+
+    //test_sequential_prime_sieve<int>();
+    //test_sequential_prime_sieve<int32_t>();
+    test_sequential_prime_sieve<int64_t>();
+    //test_sequential_prime_sieve<uint32_t>();
+    //test_sequential_prime_sieve<boost::multiprecision::cpp_int>();
+
+    //test_prime_range<int>();
+    //test_prime_range<int32_t>();
+    //test_prime_range<int64_t>();
+    //test_prime_range<uint32_t>();
```

Comment:
Why are these tests all commented out?

Reply:
`prime_range` needs to be update to use the new sieves. Since there have been such large fundamental changes I figured it was worth having outside review.